src/coreclr/jit/codegenlinear.cpp

```diff
@@ -756,6 +758,20 @@ void CodeGen::genCodeForBBlist()
 
             case BBJ_ALWAYS:
             {
+                GenTree* call = block->lastNode();
+                if ((call != nullptr) && (call->gtOper == GT_CALL))
+                {
+                    if ((call->AsCall()->gtCallMoreFlags & GTF_CALL_M_DOES_NOT_RETURN) != 0 ||
+                        ((call->AsCall()->gtCallType == CT_HELPER) &&
+                         Compiler::s_helperCallProperties.AlwaysThrow(call->AsCall()->GetHelperNum())))
+                    {
+                        // NOTE: We should probably never see a BBJ_ALWAYS block ending with a throw in a first place.
```

Comment:
I take it you did see such cases? Can you open an issues for us to follow-up and switch them over to BBJ_THROWs?

Reply:
Note this might largely clean itself up by adding the logic to `IsNoReturn`, though there is one place in morph where we still check the flag rather than the property.

Reply:
> I take it you did see such cases? 

Yes, I wanted this to be an assert, so it would not happen by accident in the future. To my surprise, the assert was hit with existing code.